### PR TITLE
[feat]: [PL-24910]: Upgrading go-scm library in scm service

### PR DESCRIPTION
**Pull request by mankrit-singh**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33032)

### CODE REVIEW

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33032)* status: 4 file(s) to review. 1 of 2 approvals obtained.

File: product/ci/scm/go.mod [hunk 1]
```diff
@@ -2,13 +2,13 @@ module github.com/harness/harness-core/product/ci/scm
 
 go 1.14
 
-replace github.com/harness/harness-core/commons/go/lib => ../../../commons/go/lib
+// replace github.com/harness/harness-core/commons/go/lib => ../../../commons/go/lib
```
Comment (shubham149) [inline product/ci/scm/go.mod:5]: undo this comment

File: product/ci/scm/go.mod [hunk 2]
```diff
@@ -2,13 +2,13 @@ module github.com/harness/harness-core/product/ci/scm
...
 
-replace github.com/harness/harness-core/product/log-service => ../../../product/log-service
+// replace github.com/harness/harness-core/product/log-service => ../../../product/log-service
```
Comment (shubham149) [inline product/ci/scm/go.mod:7]: Same here

File: product/ci/scm/go.mod.bak [hunk 3]
```diff
@@ -0,0 +1,24 @@
+module github.com/harness/harness-core/product/ci/scm
```
Comment (shubham149) [inline product/ci/scm/go.mod.bak:1]: remove this file

File: product/ci/scm/go.sum [hunk 4]
```diff
@@ -356,6 +358,7 @@ github.com/grpc-ecosystem/grpc-gateway v1.9.5/go.mod h1:vNeuVxBJEsws4ogUvrchl83t
 github.com/grpc-ecosystem/grpc-gateway v1.16.0/go.mod h1:BDjrQk3hbvj6Nolgz8mAMFbcEtjT1g+wF4CSlocrBnw=
 github.com/h2non/gock v1.0.9 h1:17gCehSo8ZOgEsFKpQgqHiR7VLyjxdAG3lkhVvO9QZU=
 github.com/h2non/gock v1.0.9/go.mod h1:CZMcB0Lg5IWnr9bF79pPMg9WeV6WumxQiUJ1UvdO1iE=
+github.com/harness/harness-core/commons/go/lib v0.0.0-20220222141117-7659b7eca599/go.mod h1:qdIXmYz6NMV8X7IV8Pfn/n/966DeC6N8GIH1x+1MkFk=
```
Comment (bhavya181) [inline product/ci/scm/go.sum:361]: @DeepakPatankar does this needs to add, because I don't see it in Mohit's earlier pr

**Review by DeepakPatankar — DISMISSED** [review]
> Reviewed 2 of 4 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33032)* status: 2 file(s) to review. 0 of 2 approvals obtained.

**Review by DeepakPatankar — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33032#-:-N1mNsF4A6wb1NYQKqwr:bnfp4nl)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`product/ci/scm/go.sum` line 361 at r2](https://reviewable.io/reviews/harness/harness-core/33032#-N1m8wAZfwxQJC_AduDX-r2-361:-N1mOvvV2Q6k_S9KMRyb:befwvtz) ([raw file](https://github.com/harness/harness-core/blob/b177db92a99f11b7d309f6f0d4d47919a28f5a72/product/ci/scm/go.sum#L361)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

@DeepakPatankar does this needs to add, because I don't see it in Mohit's earlier pr
</blockquote></details>

@shubham149 Can you check this comment

File: product/ci/scm/go.sum (hunk 4 above)
Comment (shubham149) [reply product/ci/scm/go.sum:361]: This can be removed as well. This was added due to removal of replace commands in go.mod earlier.

**Review by shubham149 — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 2 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33032)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33032)* status: 1. unresolved discussions.

**Review by mankrit-singh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33032)* status: 1. unresolved discussions. 

___
*[`product/ci/scm/go.sum` line 361 at r2](https://reviewable.io/reviews/harness/harness-core/33032#-N1m8wAZfwxQJC_AduDX-r2-361:-N1me-2w7sRWnavx9O49:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b177db92a99f11b7d309f6f0d4d47919a28f5a72/product/ci/scm/go.sum#L361)):*
<details><summary><i>Previously, shubham149 (Shubham Agrawal) wrote…</i></summary><blockquote>

This can be removed as well. This was added due to removal of replace commands in go.mod earlier.
</blockquote></details>

Done.

**Review by bhavya181 — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 2 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33032)* status: :shipit: complete! ✓

**Pull request merged**